packages/terra-hookshot/src/Hookshot.jsx

```diff
@@ -0,0 +1,336 @@
+   */
+  contentOffset: PropTypes.string,
+  /**
+   * Should content be actively positioned via hookshot.
+   */
+  isEnabled: PropTypes.bool,
+  /**
+   * Should the content be presented.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Callback function when the content have been positioned.
+   */
+  onPosition: PropTypes.func,
+  /**
+   * Required element that the content will hookshot to.
+   */
+  targetRef: PropTypes.func.isRequired,
+  /**
+   * String pair of top, middle, bottom, and left, center, right. Will mirror content is none provided.
+   */
+  targetAttachment: PropTypes.oneOf(ATTACHMENT_POSITIONS),
+  /**
+   * String pair of top and left offset, ie "10px -4px".
+   */
+  targetOffset: PropTypes.string,
+};
+
+const defaultProps = {
+  attachmentMargin: 0,
+  attachmentBehavior: 'auto',
+  contentOffset: '0 0',
+  isEnabled: false,
+  isOpen: false,
+  targetOffset: '0 0',
+};
+
+class Hookshot extends React.Component {
+  static now() {
+    return performance.now();
+  }
+
+  constructor(props) {
+    super(props);
+    this.setContentNode = this.setContentNode.bind(this);
+    this.getNodeRects = this.getNodeRects.bind(this);
+    this.update = this.update.bind(this);
+    this.tick = this.tick.bind(this);
+    this.state = { isEnabled: props.isEnabled && props.isOpen };
+    this.listenersAdded = false;
+    this.lastCall = null;
+    this.lastDuration = null;
+    this.pendingTimeout = null;
+  }
+
+  componentDidMount() {
+    if (this.state.isEnabled) {
+      if (!this.listenersAdded) {
+        this.enableListeners();
+      }
+      this.update();
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.isEnabled !== this.props.isEnabled || newProps.isOpen !== this.props.isOpen) {
+      this.setState({ isEnabled: newProps.isEnabled && newProps.isOpen });
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.isEnabled) {
+      if (!this.listenersAdded) {
+        this.enableListeners();
+      }
+      this.update();
+    } else {
+      this.disableListeners();
+    }
+  }
+
+  componentWillUnmount() {
+    this.destroy();
+  }
+
+  setContentNode(node) {
+    this.contentNode = node;
+  }
+
+  getNodeRects() {
+    const targetRect = HookshotUtils.getBounds(this.props.targetRef());
+    const contentRect = HookshotUtils.getBounds(this.contentNode);
+    const boundingRect = HookshotUtils.getBoundingRect(this.props.boundingRef ? this.props.boundingRef() : 'window');
+    return { targetRect, contentRect, boundingRect };
+  }
+
+  tick() {
+    if (this.lastDuration && this.lastDuration > 16) {
+      // Throttle to 60fps, in order to handle sarafi and mobile performance
+      this.lastDuration = Math.min(this.lastDuration - 16, 100);
+
+      // // Just in case this is the last event, remember to position just once more
+      this.pendingTimeout = setTimeout(this.tick, 100);
+      return;
+    }
+
+    if (this.lastCall && (Hookshot.now() - this.lastCall) < 10) {
```

Comment:
How did we decide on 10 out of curiosity?

Reply:
It's was just trial and error to find a happy medium as i was testing it. 10ms throws out enough without visible jittering.